src/tigerbeetle-java/src/main/java/com/tigerbeetle/Batch.java

```diff
@@ -2,43 +2,292 @@
 
 import java.nio.ByteBuffer;
 import java.nio.ByteOrder;
-import java.util.UUID;
+import static com.tigerbeetle.AssertionError.assertTrue;
 
+/**
+ * A Batch is contiguous memory block representing a colletion of elements of the same type with a
+ * cursor pointing to its current position.
+ * <p>
+ * Initially the cursor is positioned before the first element and must be positioned by calling
+ * {@link #next}, {@link #add}, or {@link #setPosition} prior to reading or writing an element.
+ */
 public abstract class Batch {
 
+    // @formatter:off
+    /*
+     * Overview
+     *
+     * Batch uses a ByteArray to hold direct memory that both the Java side and the JNI side can access.
+     *
+     * We expose the API using the concept of "Cursor" familiar to JDBC's ResultSet, where a single Batch
+     * instance points to multiple elements depending on the cursor position, eliminating individual instances
+     * for each element.
+     *
+     * Intended usage:
+     *
+     * <pre>
+     *
+     * // Populating a batch to send
+     * for(var dto : myDataSource) {
+     *
+     *   requestBatch.add();
+     *   requestBatch.setXyz(dto.Xyz);
+     *
+     * }
+     *
+     * var replyBatch = client.submit(requestBatch);
+     *
+     * // Reading from a reply
+     * while(replyBatch.next()) {
+     *
+     *   var value = replyBatch.getXyz();
```

Comment:
Again, if it's important, what is getXyz?

Reply:
I.e. if it's an example, why isn't it using real code that is relevant? Again if it's not important then you can disregard

Reply:
It's an internal overview comment block, this is intended to be an implementation comment.

I'll improve this part. Just tried to exemplify any "Batch" usage, so `getXyz` may stand for any possible `getter` implemented by one of the sub-classes, like `Account.getId` for example.